cmd/deploy/deploy.go

```diff
@@ -152,13 +152,18 @@ func Deploy(ctx context.Context) *cobra.Command {
 				}
 			}
 
+			// when name is an option - this overrides the manifest or inferred name
+			if options.Name != "" {
+				os.Setenv(model.OktetoNameEnvVar, options.Name)
```

Comment:
This seems like a very indirect way of setting this and is quite brittle. We've effectively set a global variable, in env, in order to make a decision later on. What are all of the ways this env var get set and how can we track them? Is there a more direct approach we can take that eliminates this?

It seems like a better, and less brittle way, would be to set the name as a config option and somehow passing it into the code that will eventually do the deploy directly.

Reply:
yup, this was actually not needed to be fixed :/ i've updated the scope for the PR due to the comments at the issue #2661